File: restaurant-app/src/app/model/order/restaurant-order-info.ts
import {OrderInfo} from "./order-info";
import {RestaurantShortInfo} from "../restaurant/restaurant-short-info";

export interface RestaurantOrderInfo {
  id: number;
  tableId: number;
  restaurantInfo: RestaurantShortInfo,
  orderDate: Date;
  deliveryDate: Date | any;
  orders: OrderInfo[];
  price: number;
}

File: restaurantservice/order-service/src/main/java/pl/restaurant/orderservice/api/response/RestaurantOrderInfo.java
package pl.restaurant.orderservice.api.response;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RestaurantOrderInfo {
    private Long id;
    private Long tableId;
    private RestaurantShortInfo restaurantInfo;
    private LocalDateTime orderDate;
    private LocalDateTime deliveryDate;
    private List<OrderInfo> orders;
    private BigDecimal price;
}
